### PR TITLE
snap-bootstrap, o/devicestate: use seed parallelism

### PR DESCRIPTION
**Pull request by stolowski**
Use `SetParalellism(2)` if more than 1 CPU is available; paralellism helps speed up calculation of snap  checksums. While we could use all CPUs it's not a clear win as it appears that I/O is a factor.

### CODE REVIEW

File: overlord/devicestate/firstboot.go [hunk 1]
```diff
@@ -37,7 +38,11 @@ import (
 	"github.com/snapcore/snapd/timings"
 )
 
-var errNothingToDo = errors.New("nothing to do")
+var (
+	errNothingToDo = errors.New("nothing to do")
```
Comment (mardy) [inline overlord/devicestate/firstboot.go:42]: nitpick^100: I would not mix in the same `var ()` block variables that are used just for mocking, with other variables. Or I'd add a comment to clearly identify those that are creating for being mocked.

File: overlord/devicestate/firstboot.go [hunk 2]
```diff
@@ -395,6 +400,10 @@ var loadDeviceSeed = func(st *state.State, sysLabel string) (deviceSeed seed.See
 		return nil, err
 	}
 
+	if runtimeNumCPU() > 1 {
+		deviceSeed.SetParallelism(2)
```
Comment (mardy) [inline overlord/devicestate/firstboot.go:404]: This is not tested, is it?

**Review by mardy — COMMENTED** [review]
LGTM, just a question about testing. Also, it's not clear why we set a fixed parallelism to 2 if we have more than a CPU; a comment there might help.

File: overlord/devicestate/firstboot.go (hunk 1 above)
Comment (pedronis) [reply overlord/devicestate/firstboot.go:42]: I agree on this

File: overlord/devicestate/firstboot.go (hunk 2 above)
Comment (pedronis) [reply overlord/devicestate/firstboot.go:404]: I would add some comment about:

// XXX set parallelism experimentally to 2 as I/O itself becomes a bottleneck ultimately

Comment (pedronis) [reply overlord/devicestate/firstboot.go:404]: In terms of testing we could have a test that mocks runtimeNumCPU to be 2. About checking that parallelism is set, one approach would be to look at timings and noticing the job spans?

**Review by pedronis — APPROVED** [review]
thank you, some considerations

**Review by mvo5 — APPROVED** [review]
Thank you!

File: overlord/devicestate/firstboot.go (hunk 2 above)
Comment (mvo5) [reply overlord/devicestate/firstboot.go:404]: I added the comment, I think the test can come as a followup

**Pull request merged**